### PR TITLE
HAAR-1398: upgrade springboot

### PR DESCRIPTION
**Pull request by ravmoj**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 plugins {
-  id("uk.gov.justice.hmpps.gradle-spring-boot") version "4.8.6-beta-4"
+  id("uk.gov.justice.hmpps.gradle-spring-boot") version "5.1.4-beta-4"
   kotlin("plugin.spring") version "1.8.21"
```
Comment (karenmillermoj) [inline build.gradle.kts:3]: the beta can now be removed

File: build.gradle.kts [hunk 2]
```diff
@@ -22,26 +22,24 @@ dependencies {
   implementation("org.springframework.boot:spring-boot-starter-oauth2-resource-server")
   implementation("org.springframework.boot:spring-boot-starter-oauth2-client")
   implementation("org.springframework.boot:spring-boot-starter-webflux")
-
-  implementation("uk.gov.justice.service.hmpps:hmpps-sqs-spring-boot-starter:1.2.0")
+  implementation("org.springframework.cloud:spring-cloud-starter-aws-messaging:2.2.6.RELEASE")
+  implementation("uk.gov.justice.service.hmpps:hmpps-sqs-spring-boot-starter:2.0.0-beta-15")
```
Comment (karenmillermoj) [inline build.gradle.kts:26]: Same with this one

**Review by karenmillermoj — COMMENTED** [review]
Can localstack also be updated to 1.4 ?
Not necessarily a Spring boot 3 upgrade tho, so not essential

File: build.gradle.kts (hunk 2 above)
Comment (ravmoj) [reply build.gradle.kts]: Updated

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,5 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.1.1-bin.zip
```
Comment (SimonMitchellMOJ) [inline gradle/wrapper/gradle-wrapper.properties:3]: need to stay at 8.1.1 not go back

Comment (ravmoj) [reply gradle/wrapper/gradle-wrapper.properties:3]: Reverted  to gradle-8.1.1-bin.zip

**Review by SimonMitchellMOJ — APPROVED** [review]
LGTM

**Pull request merged**